### PR TITLE
Trying for tensorflow 2.9

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address?
#1972 

##### What are the significant changes in functionality due to this change request?
Updates to tensorflow 2.9

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by joshua-cogliati-inl** [comment]
Changed tests:
From Tensorflow and related updates:

- tests/framework/MCMC/MetropolisTenDimPropMVN
- tests/framework/PostProcessors/TSACharacterizer/Basic
- tests/framework/CodeInterfaceTests/DymolaTestTimeDepNoExecutableEnsembleModel
- tests/framework/ROM/TimeSeries/ARMA/MulticycleReseed
- tests/framework/ROM/TimeSeries/SyntheticHistory/Interpolated
- tests/framework/ROM/tensorflow_keras/tf_cnn1d
- tests/framework/ROM/tensorflow_keras/tf_lstm
- tests/framework/ROM/tensorflow_keras/LSTMFromFile



From Scikitlearn update:

-    tests/framework/PostProcessors/DataMiningPostProcessor/DimensionalityReduction/SpectralEmbedding
-  tests/framework/PostProcessors/LimitSurface/testLimitSurfaceIntegralPPWithBoundingError
-   tests/framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering/MiniBatchKMeans

**Review by dylanjm — COMMENTED** [review]
Changes look good to me and changed tests have been explained in a comment above.

**Review by dylanjm — DISMISSED** [review]

**Comment by joshua-cogliati-inl** [comment]
@wangcj05  Any objections to merging this? As a warning it breaks HERON and LOGOS.  HERON is probably just an update of a pickle file (tho' @dylanjm or I should probably double check that).

File: dependencies.xml [hunk 1]
```diff
@@ -36,17 +36,17 @@ Note all install methods after "main" take
...
+    <xarray>0.20</xarray>
     <netcdf4>1.5</netcdf4>
     <matplotlib>3.2</matplotlib>
     <statsmodels>0.13</statsmodels>
-    <cloudpickle>1.6</cloudpickle>
-    <tensorflow os='windows,linux'>2.3</tensorflow>
-    <tensorflow os='mac'>2.7</tensorflow>
+    <cloudpickle>2.2</cloudpickle>
+    <tensorflow os='windows,linux'>2.9</tensorflow>
+    <tensorflow os='mac'>2.9</tensorflow>
```
Comment (wangcj05) [inline dependencies.xml:49]: Can these be combined?

**Review by wangcj05 — COMMENTED** [review]
I have one comment. I think in the future we need to have more consistent tests for tensorflow.

**Review by wangcj05 — APPROVED** [review]
PR looks good.

**Comment by wangcj05** [comment]
checklist is good, tests are green. PR can be merged.

**Comment by wangcj05** [comment]
@joshua-cogliati-inl I have merged your PR, thanks for your contribution. Here are additional requests:

- Please send an email to user group since this PR updates several libraries 

- Please send an email to plugin developer, since several of them failed due to this PR

**Pull request merged**